main.go

```diff
@@ -89,7 +89,11 @@ func main() {
 	switch {
 	case *versionFlag:
 		//if *versionFlag {
-		logger.Infof("Version: %s", version)
+		if version != nil {
+			logger.Infof("Version: %s", version)
+		} else {
+			logger.Error("Version not set correctly.")
+		}
```

Comment:
I'd refrain from this much ambiguous statement when `version` is not defined, but instead would provide a bit more insight on why. I also believe this doesn't belong in `logger`.
```suggestion
		if version != nil {
			fmt.Printf("Version: %s\n", version)
		} else {
			fmt.Println("Version not defined during build.")
		}
```